### PR TITLE
TST: prepare for freq-checking in tests.io

### PR DESCRIPTION
**Pull request by jbrockmendel**
Working on making assert_index_equal check that `freq` attrs match.  This ports the necessary test changes from tests.io.

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -226,6 +226,11 @@ def test_roundtrip_empty(self, orient, convert_axes, numpy, empty_frame):
     @pytest.mark.parametrize("numpy", [True, False])
     def test_roundtrip_timestamp(self, orient, convert_axes, numpy, datetime_frame):
         # TODO: improve coverage with date_format parameter
+
+        # freq doesnt round-trip
```
Comment (jreback) [inline pandas/tests/io/json/test_pandas.py:230]: why don’t u just change this in the fixture? tsframe

Comment (jbrockmendel) [reply pandas/tests/io/json/test_pandas.py]: updated per suggestion

Comment (jbrockmendel) [reply pandas/tests/io/json/test_pandas.py]: gentle ping; plenty more of these before were done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
these are fine

if u can yah with labels and milestone would be great

**Pull request merged**